production/template.yml

```diff
@@ -0,0 +1,321 @@
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-query
+    labels:
+      jaeger-infra: query-dc
+  spec:
+    replicas: 1
+    triggers:
+      - type: ConfigChange
+    selector:
+      name: ${JAEGER_SERVICE_NAME}-query
+    strategy:
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          name: ${JAEGER_SERVICE_NAME}-query
+      spec:
+        containers:
+        - capabilities: {}
+          image: jaegertracing/jaeger-query
+          imagePullPolicy: IfNotPresent
+          name: ${JAEGER_SERVICE_NAME}-query
+          ports:
+          - containerPort: 16686
+            protocol: TCP
+          command:
+          - "/go/bin/query-linux"
+          - "-cassandra.servers=cassandra"
+          - "-cassandra.keyspace=${KEYSPACE}"
+          - "--query.static-files=/go/jaeger-ui/"
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+    triggers:
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-query
+    labels:
+      jaeger-infra: query-service
+  spec:
+    ports:
+    - name: web
+      port: 80
+      protocol: TCP
+      targetPort: 16686
+    selector:
+      name: ${JAEGER_SERVICE_NAME}-query
+    type: ClusterIP
+- apiVersion: v1
+  kind: Route
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-query
+    labels:
+      jaeger-infra: query-route
+  spec:
+    tls:
+      insecureEdgeTerminationPolicy: Allow
+      termination: edge
+    to:
+      kind: Service
+      name: ${JAEGER_SERVICE_NAME}-query
+- apiVersion: extensions/v1beta1
+  kind: DaemonSet
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-agent
+    labels:
+      jaeger-infra: agent-daemonset
+  spec:
+    template:
+      metadata:
+        labels:
+          jaeger-infra: agent-instance
+      spec:
+        containers:
+        - name: agent-instance
+          image: jaegertracing/jaeger-agent
+          command:
+          - "/go/bin/agent-linux"
+          - "-collector.host-port=jaeger-collector:14267"
+          metadata:
+            labels:
+              jaeger-infra: agent-instance
+          ports:
+          - containerPort: 5775
+            protocol: UDP
+          - containerPort: 6831
+            protocol: UDP
+          - containerPort: 6832
+            protocol: UDP
+- apiVersion: v1
+  kind: Service
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-agent
+    labels:
+      jaeger-infra: agent-service
+  spec:
+    ports:
+    - name: zipkin-model
+      port: 5775
+      protocol: UDP
+      targetPort: 5775
+    - name: jaeger-model-compact
+      port: 6831
+      protocol: UDP
+      targetPort: 6831
+    - name: jaeger-model-binary
+      port: 6832
+      protocol: UDP
+      targetPort: 6832
+    clusterIP: None
+    selector:
+      jaeger-infra: agent-instance
+parameters:
```

Comment:
When you open the file you see all parameters at the top, it's like with a function/class/script, you also define parameters at the beginning. I don't know how people write  these templates maybe they do it like you did.